database/databricks/sql_warehouse.go

```diff
@@ -0,0 +1,228 @@
+//go:build go1.9
+// +build go1.9
+
+package databricks
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	_ "github.com/databricks/databricks-sql-go"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/golang-migrate/migrate/v4/database/multistmt"
+	"go.uber.org/atomic"
+)
+
+const RegistrationKey = "databricks-sqlwarehouse"
+
+func init() {
+	db := SQLWarehouse{}
+	database.Register(RegistrationKey, &db)
+}
+
+var (
+	multiStmtDelimiter = []byte(";")
+
+	DefaultMigrationsTable       = "schema_migrations"
```

Comment:
As in my notes, we could and probably should point this at `catalog_name`.schema_name.schema_migrations before merging.